File: solace/cmd/cooldown/Skullknock.java
package solace.cmd.cooldown;

import java.util.*;
import solace.game.*;
import solace.net.*;
import java.io.*;
import solace.util.*;

/**
 * A potency 150 attack that stuns the target for 4 seconds.
 * @author Ryan Sandor Richards
 */
public class Skullknock extends CooldownCommand {
  public Skullknock(Player p) {
    super("skullknock", p);
    setCooldownDuration(180);
    setBasePotency(150);
    setInitiatesCombat(true);
  }

  public boolean execute(int level, Player target) {
    boolean isHit = executeAttack(target);
    if (isHit) {
      target.applyBuff(Buffs.create("stunned", 4));
    }
    return isHit;
  }
}

File: script/cooldowns/skullknock.js
'use strict';

var Buffs = Packages.solace.util.Buffs;

/**
 * A potency 150 attack that stuns the target for 4 seconds.
 * @author Ryan Sandor Richards
 */
Commands.addCooldown('skullknock', {
  cooldownDuration: 180,
  initiatesCombat: true,
  basePotency: 150,
  run: function (level, player, target, cooldown) {
    if (cooldown.executeAttack(target)) {
      target.applyBuff(Buffs.create("stunned", 4));
    }
  }
});
